Apps/Admin/Tests/Functional/cypress/integration/e2e/agentaccess.cy.js

```diff
@@ -68,37 +68,43 @@ describe("Provision", () => {
     });
 
     it("Edit User", () => {
+        cy.intercept("GET", `**/AgentAccess/?query=${user}`).as("getUser");
         cy.get("[data-testid=query-input]").clear().type(user);
         cy.get("[data-testid=search-btn]").click();
         cy.get("[data-testid^=agent-table-username-]")
             .contains(user.toLowerCase())
             .parents(".mud-table-row")
             .get("[data-testid^=agent-table-edit-btn]")
             .click();
-        cy.get("[data-testid=provision-dialog-modal-text]").should(
-            "be.visible"
-        );
-        cy.get("[data-testid=roles-select]").click();
-        cy.get("[data-testid=role]").contains("AdminAnalyst").click();
-        cy.get("[data-testid=save-btn]").parent().parent().click(0, 0);
-        cy.get("[data-testid=save-btn]").click();
-        cy.get("[data-testid=provision-dialog-modal-text]").should("not.exist");
+        cy.wait("@getUser").then(() => {
```

Comment:
Later commands [do not need to be nested](https://docs.cypress.io/api/commands/wait#Nesting).

Also note that the `wait()` command will time out just as easily as others, so if the time allocated is currently insufficient, you may need to pass in a larger timeout value.